### PR TITLE
feat(masir): overhaul of core logic

### PR DESCRIPTION
**Pull request by lukeyou05**
This pull request overhauls much of _masir_'s core logic to improve overall functionality. It can be summarized as follows:
1. _masir_ now additionally calls `GetAncestor()` to grab the top-level/root hwnd under the cursor, as opposed to directly using the hwnd from `WindowFromPoint()`, which might return a child window. This ensures correct behavior when raising windows to the foreground, which generally should be a top-level window.
2. _masir_ now tracks mouse press/release to avoid raising any windows while the mouse is being held down (i.e. when resizing a window or dragging and dropping files).
3. _masir_ now checks window styles to help determine eligibility when running without any twm integration. This means _masir_ is now completely standalone, whereas previously it had a soft dependency on komorebi and its hwnds file.
5. As a result of the above change, komorebi integration is no longer automatic; it has to be specified with the `--komorebi` argument when _masir_ is launched from the terminal.
7. _masir_ now checks **BOTH** the eligibility of the cursor hwnd as well as the foreground hwnd before raising any windows. This was done to improve compatibility with applications like Flow Launcher or PowerToys' PowerLauncher which previously would close as soon as they were launched if the cursor was outside their window area.
9. The call to `SetWindowPos` when raising windows was removed. It caused an issue where my auto-hidden taskbar would sometimes pop up when switching komorebi workspaces, and _masir_ seems to work fine without it from my testing.

TL;DR (the most important changes): 
1. _masir_ now requires the `--komorebi` argument to enable komorebi integration.
2. _masir_ now checks **BOTH** the eligibility of the cursor hwnd as well as the foreground hwnd.
3. Overall, _masir_ should work better now (hopefully!)

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -10,38 +10,47 @@ use windows::Win32::Foundation::POINT;
...
+    Contains,
+    Equals,
+}
 
 #[derive(Parser)]
 #[clap(author, about, version)]
 struct Opts {
+    /// Enable komorebi integration to avoid raising unmanaged windows
+    #[clap(long)]
+    komorebi: bool,
```
Comment (LGUG2Z) [inline src/main.rs:53]: Let's switch this to a negative boolean like `--disable-integrations` because any future integration can provide a HWNDs file, and this will also avoid breaking changes in the CLI between versions

Comment (lukeyou05) [reply src/main.rs]: Good idea! Perhaps we could also have komorebi (and other TWMs in the future) automatically deletes their HWNDs file upon exiting? This would make it easier to select between multiple integrations with other tiling window managers, and would also automatically disable integrations when no TWMs are running.

Comment (LGUG2Z) [reply src/main.rs]: Ah I wish; we need to keep the HWNDs file around for komorebi at least to give the user a fallback way of restoring the windows (`komorebic restore-windows`) if for whatever reason the process is terminated in a way that doesn't allow the cleanup hooks which restore hidden windows to run properly 😞
